packages/react/src/ReactHooks.js

```diff
@@ -81,6 +90,15 @@ export function useReducer<S, I, A>(
 
 export function useRef<T>(initialValue: T): {current: T} {
   const dispatcher = resolveDispatcher();
+  if (__DEV__) {
+    // this is the only place where, if the user gives no arguments,
+    // the argument will actually be missing instead of just being undefined.
+    warning(
+      arguments.length === 1,
+      'Expected exactly one argument to React.useRef(initialValue), instead received %s.',
+      arguments.length,
+    );
+  }
   return dispatcher.useRef(initialValue);
```

Comment:
I thought of adding the warnings to `mountRef` / `mountState` instead but, because of how the dispatcher is invoked, by the time things reach them the `arguments.length` will already be `1`. `ReactHooks.js` is the only place that sees the `arguments` exactly as provided by the user.